File: src/types/generator.ts
export type GeneratorModuleId = 'uuid' | 'nanoid' | 'random' | 'password' | 'apiKey' | 'hash' | 'jwt' | 'timestamp';
export type HashAlgorithm = 'md5' | 'sha1' | 'sha256' | 'sha512';
export type RandomValueMode = 'integer' | 'float' | 'string';

export interface GeneratorItemsResult {
  text: string;
  items: string[];
  meta: string[];
}

export interface HashGenerateResult {
  algorithm: string;
  value: string;
  meta: string[];
}

export interface JwtGenerateResult {
  token: string;
  headerPretty: string;
  payloadPretty: string;
  meta: string[];
}

export interface JwtDecodeResult {
  headerPretty: string;
  payloadPretty: string;
  signature: string;
  meta: string[];
}

export interface UuidGenerateOptions {
  count: number;
  uppercase: boolean;
  removeHyphens: boolean;
}

export interface NanoIdGenerateOptions {
  length: number;
  count: number;
  alphabet?: string | null;
}

export interface RandomValueGenerateOptions {
  mode: RandomValueMode;
  count: number;
  min: number;
  max: number;
  decimalPlaces: number;
  length: number;
  charset?: string | null;
}

export interface PasswordGenerateOptions {
  length: number;
  count: number;
  includeUppercase: boolean;
  includeLowercase: boolean;
  includeNumbers: boolean;
  includeSymbols: boolean;
  excludeSimilar: boolean;
}

export interface ApiKeyGenerateOptions {
  prefix: string;
  length: number;
  count: number;
  separator: string;
}

export interface HashGenerateOptions {
  algorithm: HashAlgorithm;
  input: string;
}

export interface JwtGenerateOptions {
  payloadJson: string;
  secret: string;
  headerJson: string;
}

export interface JwtDecodeOptions {
  token: string;
}

export const GENERATOR_MODULES: Array<{ id: GeneratorModuleId; label: string; description: string }> = [
  { id: 'uuid', label: 'UUID', description: '批量生成 UUID / GUID。' },
  { id: 'nanoid', label: 'NanoID', description: '生成更短的随机 ID。' },
  { id: 'random', label: '随机值', description: '生成整数、浮点数或随机字符串。' },
  { id: 'password', label: '密码', description: '按规则生成测试密码。' },
  { id: 'apiKey', label: 'API Key', description: '生成带前缀的 token/key。' },
  { id: 'hash', label: 'Hash', description: '计算 MD5 / SHA 摘要。' },
  { id: 'jwt', label: 'JWT', description: '生成或解析本地 mock JWT。' },
  { id: 'timestamp', label: '时间戳', description: '快捷生成当前秒级与毫秒级时间戳。' },
];

File: src-tauri/src/models/generator.rs
//! 生成器相关数据模型
//!
//! 定义生成器中心使用的数据结构

use serde::{Deserialize, Serialize};

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub enum HashAlgorithm {
    Md5,
    Sha1,
    Sha256,
    Sha512,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub enum RandomValueMode {
    Integer,
    Float,
    String,
}

#[derive(Debug, Clone, Copy, Serialize, Deserialize, PartialEq, Eq)]
pub enum CountryPreset {
    #[serde(rename = "cn")]
    Cn,
    #[serde(rename = "us")]
    Us,
    #[serde(rename = "uk")]
    Uk,
    #[serde(rename = "jp")]
    Jp,
    #[serde(rename = "de")]
    De,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UuidGenerateOptions {
    #[serde(default = "default_batch_count")]
    pub count: usize,
    #[serde(default)]
    pub uppercase: bool,
    #[serde(default)]
    pub remove_hyphens: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct NanoIdGenerateOptions {
    #[serde(default = "default_length_21")]
    pub length: usize,
    #[serde(default = "default_batch_count")]
    pub count: usize,
    #[serde(default)]
    pub alphabet: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct RandomValueGenerateOptions {
    pub mode: RandomValueMode,
    #[serde(default = "default_batch_count")]
    pub count: usize,
    #[serde(default = "default_min")]
    pub min: i64,
    #[serde(default = "default_max")]
    pub max: i64,
    #[serde(default = "default_decimal_places")]
    pub decimal_places: u8,
    #[serde(default = "default_length_16")]
    pub length: usize,
    #[serde(default)]
    pub charset: Option<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PasswordGenerateOptions {
    #[serde(default = "default_length_16")]
    pub length: usize,
    #[serde(default = "default_batch_count")]
    pub count: usize,
    #[serde(default = "default_true")]
    pub include_uppercase: bool,
    #[serde(default = "default_true")]
    pub include_lowercase: bool,
    #[serde(default = "default_true")]
    pub include_numbers: bool,
    #[serde(default = "default_true")]
    pub include_symbols: bool,
    #[serde(default)]
    pub exclude_similar: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct ApiKeyGenerateOptions {
    #[serde(default)]
    pub prefix: String,
    #[serde(default = "default_length_24")]
    pub length: usize,
    #[serde(default = "default_batch_count")]
    pub count: usize,
    #[serde(default = "default_separator")]
    pub separator: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HashGenerateOptions {
    pub algorithm: HashAlgorithm,
    pub input: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct JwtGenerateOptions {
    pub payload_json: String,
    #[serde(default)]
    pub secret: String,
    #[serde(default)]
    pub header_json: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct JwtDecodeOptions {
    pub token: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UserDataGenerateOptions {
    pub country: CountryPreset,
    #[serde(default = "default_batch_count")]
    pub count: usize,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct IdentityGenerateOptions {
    pub country: CountryPreset,
    #[serde(default = "default_batch_count")]
    pub count: usize,
    #[serde(default)]
    pub document_type: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct PaymentCardGenerateOptions {
    pub country: CountryPreset,
    #[serde(default = "default_batch_count")]
    pub count: usize,
    #[serde(default)]
    pub brand: String,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UserProfileGenerateOptions {
    pub country: CountryPreset,
    #[serde(default = "default_batch_count")]
    pub count: usize,
    #[serde(default = "default_true")]
    pub include_profile: bool,
    #[serde(default = "default_true")]
    pub include_contact: bool,
    #[serde(default = "default_true")]
    pub include_address: bool,
    #[serde(default = "default_true")]
    pub include_company: bool,
    #[serde(default)]
    pub include_identity: bool,
    #[serde(default)]
    pub include_payment: bool,
    #[serde(default)]
    pub include_account: bool,
    #[serde(default)]
    pub include_preferences: bool,
    #[serde(default)]
    pub include_device: bool,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct GeneratorItemsResult {
    pub text: String,
    pub items: Vec<String>,
    pub meta: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct HashGenerateResult {
    pub algorithm: String,
    pub value: String,
    pub meta: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct JwtGenerateResult {
    pub token: String,
    pub header_pretty: String,
    pub payload_pretty: String,
    pub meta: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct JwtDecodeResult {
    pub header_pretty: String,
    pub payload_pretty: String,
    pub signature: String,
    pub meta: Vec<String>,
}

#[derive(Debug, Clone, Serialize, Deserialize)]
#[serde(rename_all = "camelCase")]
pub struct UserDataGenerateResult {
    pub country: String,
    pub locale_label: String,
    pub text: String,
    pub items: Vec<String>,
    pub json: String,
    pub meta: Vec<String>,
}

fn default_batch_count() -> usize {
    1
}

fn default_length_21() -> usize {
    21
}

fn default_length_16() -> usize {
    16
}

fn default_length_24() -> usize {
    24
}

fn default_min() -> i64 {
    0
}

fn default_max() -> i64 {
    1000
}

fn default_decimal_places() -> u8 {
    2
}

fn default_true() -> bool {
    true
}

fn default_separator() -> String {
    "_".to_string()
}
